libs/rapidsandrivers-extras/src/main/kotlin/rapidsandrivers/migrering/ListenerMedLogging.kt

```diff
@@ -4,9 +4,23 @@ import no.nav.etterlatte.libs.common.logging.withLogContext
 import no.nav.etterlatte.libs.common.rapidsandrivers.correlationId
 import no.nav.helse.rapids_rivers.JsonMessage
 import no.nav.helse.rapids_rivers.MessageContext
+import no.nav.helse.rapids_rivers.RapidsConnection
 import no.nav.helse.rapids_rivers.River
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
 
-abstract class ListenerMedLogging : River.PacketListener {
+abstract class ListenerMedLogging(private val rapidsConnection: RapidsConnection) : River.PacketListener {
+
+    protected val logger: Logger = LoggerFactory.getLogger(this::class.java)
+    fun initialiser(spesifikkInitialisering: River.() -> Unit) {
+        val navn = this.javaClass.simpleName
+        logger.info("Initialiserer rapid for $navn")
+        River(rapidsConnection).apply {
+            correlationId()
+            spesifikkInitialisering()
+        }.register(this)
+        logger.info("Initialisert ferdig rapid for $navn")
+    }
```

Comment:
Hva med å bruke abstraksjon for å fjerne behovet for "manuell" init i hver eneste klasse? 
Eksempelvis noe i denne gaten: 

```
    init {
        registerRiver()
    }

    private fun registerRiver() {
        val navn = this.javaClass.simpleName
        logger.info("Initialiserer rapid for $navn")
        River(rapidsConnection).apply {
            correlationId()
            eventName(eventName)
            validation()
        }.register(this)
        logger.info("Initialisert ferdig rapid for $navn")
    }

    abstract val eventName: String
    abstract fun River.validation()
```

Så tvinges utvikleren til å ta stilling til `eventName` og `validation`... som her: 

```
internal class JournalfoerVedtaksbrev(
    private val rapidsConnection: RapidsConnection,
    private val service: VedtaksbrevService
) : RiverMedLogging(rapidsConnection) {

    override val eventName = KafkaHendelseType.ATTESTERT.toString()

    override fun River.validation() {
        validate { it.requireKey("vedtak") }
        validate { it.requireKey("vedtak.vedtakId") }
        validate { it.requireKey("vedtak.behandling.id") }
        validate { it.requireKey("vedtak.sak") }
        validate { it.requireKey("vedtak.sak.id") }
        validate { it.requireKey("vedtak.sak.ident") }
        validate { it.requireKey("vedtak.sak.sakType") }
        validate { it.requireKey("vedtak.vedtakFattet.ansvarligEnhet") }
        validate {
            it.rejectValues("vedtak.behandling.type", listOf(BehandlingType.MANUELT_OPPHOER.name))
        }
        validate { it.rejectValue(SKAL_SENDE_BREV, false) }
    }
```

Reply:
Ja, liker tanken. Prøvde noko sånt først, men da med init- og registerRiver-blokkene slått saman, og da ga IntelliJ åtvaring på register(this):
```
River(rapidsConnection).apply {
            correlationId()
            eventName(eventName)
            validation()
        }.register(this)
```
:

`Leaking 'this' in constructor of non-final class RiverMedLogging 
 Inspection info: Reports unsafe operations with this during object construction including:
Accessing a non-final property during class initialization: from a constructor or property initialization
Calling a non-final function during class initialization
Using this as a function argument in a constructor of a non-final class
If other classes inherit from the given class, they may not be fully initialized at the moment when an unsafe operation is carried out.
Example:
  abstract class Base {
      val code = calculate()
      abstract fun calculate(): Int
  }

  class Derived(private val x: Int) : Base() {
      override fun calculate() = x
  }

  fun testIt() {
      println(Derived(42).code) // Expected: 42, actual: 0
  }`

Så litt usikker på om det er trygt. init-blokka i superklassa blir om eg har skjønt det rett køyrd før init og anna initialisering i den konkrete klassa.

Samtidig ser det ut som det einaste som skjer i register-metoden at listeneren blir lagt til i ei liste over kven som lyttar, så det skal jo mykje til at det blir feil av det. Hadde jo vore veldig fint å få det på plass, så kanskje risikoen er verdt det 🤔

Reply:
Kikka på https://medium.com/keepsafe-engineering/an-in-depth-look-at-kotlins-initializers-a0420fcbf546 , og ser at det vil bli litt knot å få det til, så eg let ideen kverne litt, lurer på om det går an å bruke interface på eit fornuftig vis også, men tar det i så fall i ein eigen PR.